subsys/zigbee/cli/zigbee_cli_ctx_mgr.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZIGBEE_CLI_CTX_MGR_H__
+#define ZIGBEE_CLI_CTX_MGR_H__
+
+#include <shell/shell.h>
+
+#include "zigbee_cli_zdo_types.h"
+
+#define CTX_MGR_ENTRY_IVALID_INDEX 0xFF
+
+
+/* Type of a frame a context manager entry is associated with. */
+enum ctx_entry_type {
+	CTX_MGR_INVALID_ENTRY_TYPE,
+	CTX_MGR_ZDO_ENTRY_TYPE,
+	CTX_MGR_ZCL_ENTRY_TYPE,
+	CTX_MGR_PING_ENTRY_TYPE,
+};
+
+/* A context manager entry structure associated with a given frame,
+ * used to store a frame related data and a shell to be used for logging a data.
+ * Frame is identified by the type and its unique identifier number:
+ *     - ZDO frames are using the transaction number (TSN) as identifier number.
+ *     - ZCL frames are identified using the ZCL Sequence number.
+ *     - PING frames are using the custom counter, implemented along
+ *       the PING commands.
+ */
+struct ctx_entry {
+	bool taken;
+	/* Unique identifier of a frame. Type of used identifier depends on
+	 * the type of frame the entry is associated with.
+	 */
+	uint8_t id;
+	enum ctx_entry_type type;
+	/* Pointer to the shell to be used for printing logs. */
+	const struct shell *shell;
+	/* Union of Structures used to store a frame related data,
+	 * shared between the different frame types.
+	 */
+	union {
+		struct zdo_data zdo_data;
+	};
+};
+
+/**@brief Delete context manager entry to mark it free and prepare to be
+ *        reused later.
+ *
+ * @param ctx_entry       Pointer to the context manager entry to be deleted.
+ */
+void ctx_mgr_delete_entry(struct ctx_entry *ctx_entry);
+
+/**@brief Get a pointer to a unused context manager entry.
+ *
+ * @param type            Type of a data to be stored in an entry.
```

Comment:
Not a big deal, but I don't see a reason to add such a long spacings before the actual description, IMO it's just harder to read.